core/trino-main/src/main/java/io/trino/sql/planner/PlanOptimizers.java

```diff
@@ -700,15 +700,14 @@ public PlanOptimizers(
                         estimatedExchangesCostCalculator,
                         ImmutableSet.of(new RemoveRedundantIdentityProjections())),
                 // Prefer write partitioning rule requires accurate stats.
-                // Therefore PredicatePushDown, columnPruningOptimizer and
-                // RemoveRedundantIdentityProjections need to run beforehand.
+                // Therefore PredicatePushDown and columnPruningOptimizer
+                // need to run beforehand.
                 new IterativeOptimizer(
                         ruleStats,
                         statsCalculator,
                         estimatedExchangesCostCalculator,
                         ImmutableSet.of(new PreferWritePartitioning())),
-                // Because ReorderJoins runs only once,
-                // PredicatePushDown, columnPruningOptimizer and RemoveRedundantIdentityProjections
```

Comment:
What's wrong with this comment?

Reply:
Indeed. `RemoveRedundantIdentityProjections` was actually needed so that CBO could construct muli-join node (I added support for projections in CBO)